sqlmesh/core/scheduler.py

```diff
@@ -55,9 +56,43 @@
 
 logger = logging.getLogger(__name__)
 SnapshotToIntervals = t.Dict[Snapshot, Intervals]
-# we store snapshot name instead of snapshots/snapshotids because pydantic
-# is extremely slow to hash. snapshot names should be unique within a dag run
-SchedulingUnit = t.Tuple[str, t.Tuple[Interval, int]]
+
+
+class SchedulingUnit(abc.ABC):
+    snapshot_name: str
```

Comment:
If you make this a dataclass then you don't have to redefine `snapshot_name` in the classes that inherit from this.

Reply:
I think when we talked you said you wanted this to be more of an interface but I don't really get the advantage we get from this framing. Either way works though.